// @ng-bootstrap/ng-bootstrap/datepicker/datepicker.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from './datepicker-day-view.ngfactory';
import * as import2 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker-day-view';
import * as import3 from './datepicker-navigation.ngfactory';
import * as import4 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker-navigation';
import * as import5 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker-i18n';
import * as import6 from '@ng-bootstrap/ng-bootstrap/datepicker/ngb-calendar';
import * as import7 from '@angular/common';
import * as import8 from './datepicker-month-view.ngfactory';
import * as import9 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker-month-view';
import * as import10 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker';
import * as import11 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker-service';
import * as import12 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker-config';
import * as import13 from '@angular/forms';
const styles_NgbDatepicker:any[] = ['[_nghost-%COMP%] {\n      border: 1px solid rgba(0, 0, 0, 0.125);\n    }\n    .ngb-dp-header[_ngcontent-%COMP%] {\n      border-bottom: 1px solid rgba(0, 0, 0, 0.125);\n    }\n    .ngb-dp-month[_ngcontent-%COMP%] {\n      pointer-events: none;\n    }\n    ngb-datepicker-month-view[_ngcontent-%COMP%] {\n      pointer-events: auto;\n    }\n    .ngb-dp-month[_ngcontent-%COMP%]:first-child {\n      margin-left: 0 !important;\n    }    \n    .ngb-dp-month-name[_ngcontent-%COMP%] {\n      font-size: larger;\n      height: 2rem;\n      line-height: 2rem;\n    }'];
export const RenderType_NgbDatepicker:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 0,
  styles: styles_NgbDatepicker,
  data: {}
}
);
function View_NgbDatepicker_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n       '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'ngbDatepickerDayView',
        ''
      ]
    ],[
      [
        2,
        'bg-primary',
        (null as any)
      ]
      ,
      [
        2,
        'text-white',
        (null as any)
      ]
      ,
      [
        2,
        'text-muted',
        (null as any)
      ]
      ,
      [
        2,
        'outside',
        (null as any)
      ]
      ,
      [
        2,
        'btn-secondary',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import1.View_NgbDatepickerDayView_0,import1.RenderType_NgbDatepickerDayView)),
    import0.ɵdid(49152,(null as any),0,import2.NgbDatepickerDayView,([] as any[]),{
      currentMonth: [
        0,
        'currentMonth'
      ]
      ,
      date: [
        1,
        'date'
      ]
      ,
      disabled: [
        2,
        'disabled'
      ]
      ,
      selected: [
        3,
        'selected'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    const currVal_5:any = v.context.currentMonth;
    const currVal_6:any = v.context.date;
    const currVal_7:any = v.context.disabled;
    const currVal_8:any = v.context.selected;
    ck(v,2,0,currVal_5,currVal_6,currVal_7,currVal_8);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,2).selected;
    const currVal_1:any = import0.ɵnov(v,2).selected;
    const currVal_2:any = import0.ɵnov(v,2).isMuted();
    const currVal_3:any = import0.ɵnov(v,2).isMuted();
    const currVal_4:boolean = !import0.ɵnov(v,2).disabled;
    ck(v,1,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
function View_NgbDatepicker_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'ngb-datepicker-navigation',[[
        'class',
        'd-flex justify-content-between'
      ]
      ],[[
        2,
        'collapsed',
        (null as any)
      ]
    ],[
      [
        (null as any),
        'navigate'
      ]
      ,
      [
        (null as any),
        'select'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('navigate' === en)) {
        const pd_0:any = ((<any>co.onNavigateEvent($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('select' === en)) {
        const pd_1:any = ((<any>co.onNavigateDateSelect($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },import3.View_NgbDatepickerNavigation_0,import3.RenderType_NgbDatepickerNavigation)),
    import0.ɵdid(49152,(null as any),0,import4.NgbDatepickerNavigation,[
      import5.NgbDatepickerI18n,
      import6.NgbCalendar
    ]
    ,{
      date: [
        0,
        'date'
      ]
      ,
      disabled: [
        1,
        'disabled'
      ]
      ,
      maxDate: [
        2,
        'maxDate'
      ]
      ,
      minDate: [
        3,
        'minDate'
      ]
      ,
      months: [
        4,
        'months'
      ]
      ,
      showSelect: [
        5,
        'showSelect'
      ]
      ,
      showWeekNumbers: [
        6,
        'showWeekNumbers'
      ]

    }
    ,{
      navigate: 'navigate',
      select: 'select'
    }
    ),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = ((co.months[0] == null)? (null as any): co.months[0].firstDate);
    const currVal_2:any = co.disabled;
    const currVal_3:any = co._maxDate;
    const currVal_4:any = co._minDate;
    const currVal_5:any = co.months.length;
    const currVal_6:any = (co.navigation === 'select');
    const currVal_7:any = co.showWeekNumbers;
    ck(v,1,0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7);
  },(ck,v) => {
    const currVal_0:boolean = !import0.ɵnov(v,1).showSelect;
    ck(v,0,0,currVal_0);
  });
}
function View_NgbDatepicker_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'ngb-dp-month-name text-center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n            ',
      ' ',
      '\n          '
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.i18n.getMonthFullName((<any>v.parent).context.$implicit.number);
    const currVal_1:any = (<any>v.parent).context.$implicit.year;
    ck(v,1,0,currVal_0,currVal_1);
  });
}
function View_NgbDatepicker_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),8,'div',[[
        'class',
        'ngb-dp-month d-block ml-3'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['            \n          '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbDatepicker_4)),
    import0.ɵdid(16384,(null as any),0,import7.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n          '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),2,'ngb-datepicker-month-view',[[
        'class',
        'd-block'
      ]
      ],(null as any),[[
        (null as any),
        'select'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('select' === en)) {
        const pd_0:any = ((<any>co.onDateSelect($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import8.View_NgbDatepickerMonthView_0,import8.RenderType_NgbDatepickerMonthView)),
    import0.ɵdid(49152,(null as any),0,import9.NgbDatepickerMonthView,[import5.NgbDatepickerI18n],{
      dayTemplate: [
        0,
        'dayTemplate'
      ]
      ,
      disabled: [
        1,
        'disabled'
      ]
      ,
      month: [
        2,
        'month'
      ]
      ,
      outsideDays: [
        3,
        'outsideDays'
      ]
      ,
      selectedDate: [
        4,
        'selectedDate'
      ]
      ,
      showWeekdays: [
        5,
        'showWeekdays'
      ]
      ,
      showWeekNumbers: [
        6,
        'showWeekNumbers'
      ]

    }
    ,{select: 'select'}),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ((co.navigation !== 'select') || (co.displayMonths > 1));
    ck(v,4,0,currVal_0);
    const currVal_1:any = (co.dayTemplate || import0.ɵnov((<any>v.parent),1));
    const currVal_2:any = co.disabled;
    const currVal_3:any = v.context.$implicit;
    const currVal_4:any = ((co.displayMonths === 1)? co.outsideDays: 'hidden');
    const currVal_5:any = co.model;
    const currVal_6:any = co.showWeekdays;
    const currVal_7:any = co.showWeekNumbers;
    ck(v,7,0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7);
  },(null as any));
}
export function View_NgbDatepicker_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵand(0,[[
        'dt',
        2
      ]
    ],(null as any),0,(null as any),View_NgbDatepicker_1)),
    (l()(),import0.ɵted((null as any),['\n    \n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'ngb-dp-header bg-faded pt-1 rounded-top'
      ]
    ],[
      [
        4,
        'height',
        'rem'
      ]
      ,
      [
        4,
        'marginBottom',
        'rem'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbDatepicker_2)),
    import0.ɵdid(16384,(null as any),0,import7.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'ngb-dp-months d-flex px-1 pb-1'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbDatepicker_3)),
    import0.ɵdid(802816,(null as any),0,import7.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import10.NgbDatepicker = v.component;
    const currVal_2:any = (co.navigation !== 'none');
    ck(v,6,0,currVal_2);
    const currVal_3:any = co.months;
    ck(v,12,0,currVal_3);
  },(ck,v) => {
    var co:import10.NgbDatepicker = v.component;
    const currVal_0:any = co.getHeaderHeight();
    const currVal_1:any = (0 - co.getHeaderMargin());
    ck(v,3,0,currVal_0,currVal_1);
  });
}
function View_NgbDatepicker_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),3,'ngb-datepicker',[[
        'class',
        'd-inline-block rounded'
      ]
    ],(null as any),(null as any),(null as any),View_NgbDatepicker_0,RenderType_NgbDatepicker)),
    import0.ɵprd(512,(null as any),import11.NgbDatepickerService,import11.NgbDatepickerService,[import6.NgbCalendar]),
    import0.ɵdid(638976,(null as any),0,import10.NgbDatepicker,[
      import11.NgbDatepickerService,
      import6.NgbCalendar,
      import5.NgbDatepickerI18n,
      import12.NgbDatepickerConfig
    ]
    ,(null as any),(null as any)),
    import0.ɵprd(5120,(null as any),import13.NG_VALUE_ACCESSOR,(p0_0:any) => {
      return [p0_0];
    },[import10.NgbDatepicker])
  ]
  ,(ck,v) => {
    ck(v,2,0);
  },(null as any));
}
export const NgbDatepickerNgFactory:import0.ComponentFactory<import10.NgbDatepicker> = import0.ɵccf('ngb-datepicker',import10.NgbDatepicker,View_NgbDatepicker_Host_0,{
  dayTemplate: 'dayTemplate',
  displayMonths: 'displayMonths',
  firstDayOfWeek: 'firstDayOfWeek',
  markDisabled: 'markDisabled',
  minDate: 'minDate',
  maxDate: 'maxDate',
  navigation: 'navigation',
  outsideDays: 'outsideDays',
  showWeekdays: 'showWeekdays',
  showWeekNumbers: 'showWeekNumbers',
  startDate: 'startDate'
}
,{navigate: 'navigate'},([] as any[]));
