// @ng-bootstrap/ng-bootstrap/datepicker/datepicker-month-view.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker-month-view';
import * as import3 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker-i18n';
const styles_NgbDatepickerMonthView:any[] = ['.ngb-dp-weekday[_ngcontent-%COMP%], .ngb-dp-week-number[_ngcontent-%COMP%] {\n      line-height: 2rem;\n    }\n    .ngb-dp-day[_ngcontent-%COMP%], .ngb-dp-weekday[_ngcontent-%COMP%], .ngb-dp-week-number[_ngcontent-%COMP%] {\n      width: 2rem;\n      height: 2rem;      \n    }\n    .ngb-dp-day[_ngcontent-%COMP%] {\n      cursor: pointer;\n    }\n    .ngb-dp-day.disabled[_ngcontent-%COMP%], .ngb-dp-day.hidden[_ngcontent-%COMP%] {\n      cursor: default;\n    }'];
export const RenderType_NgbDatepickerMonthView:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 0,
  styles: styles_NgbDatepickerMonthView,
  data: {}
}
);
function View_NgbDatepickerMonthView_2(l:any):import0.ɵViewDefinition {
      return import0.ɵvid(0,[(l()(),import0.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'ngb-dp-weekday'
      ]
  ],(null as any),(null as any),(null as any),(null as any),(null as any)))],(null as any),(null as any));
}
function View_NgbDatepickerMonthView_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'ngb-dp-weekday small text-center text-info font-italic'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n        ',
      '\n      '
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.i18n.getWeekdayShortName(v.context.$implicit);
    ck(v,1,0,currVal_0);
  });
}
function View_NgbDatepickerMonthView_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'ngb-dp-week d-flex'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbDatepickerMonthView_2)),
    import0.ɵdid(16384,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbDatepickerMonthView_3)),
    import0.ɵdid(802816,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.showWeekNumbers;
    ck(v,3,0,currVal_0);
    const currVal_1:any = co.month.weekdays;
    ck(v,6,0,currVal_1);
  },(null as any));
}
function View_NgbDatepickerMonthView_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'ngb-dp-week-number small text-center font-italic text-muted'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = (<any>(<any>v.parent).parent).context.$implicit.number;
    ck(v,1,0,currVal_0);
  });
}
function View_NgbDatepickerMonthView_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n            ']))],(null as any),(null as any));
}
function View_NgbDatepickerMonthView_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n            '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),3,(null as any),View_NgbDatepickerMonthView_9)),
    import0.ɵdid(540672,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod([
      'year',
      'month',
      'day'
    ]
    ),
    import0.ɵpod([
      'date',
      'currentMonth',
      'disabled',
      'selected'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n          ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.dayTemplate;
    const currVal_1:any = ck(v,4,0,ck(v,3,0,(<any>v.parent).context.$implicit.date.year,(<any>v.parent).context.$implicit.date.month,(<any>v.parent).context.$implicit.date.day),co.month.number,co.isDisabled((<any>v.parent).context.$implicit),co.isSelected((<any>v.parent).context.$implicit.date));
    ck(v,2,0,currVal_0,currVal_1);
  },(null as any));
}
function View_NgbDatepickerMonthView_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'ngb-dp-day'
      ]
    ],[
      [
        2,
        'disabled',
        (null as any)
      ]
      ,
      [
        2,
        'hidden',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.doSelect(v.context.$implicit)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n          '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbDatepickerMonthView_8)),
    import0.ɵdid(16384,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:boolean = !co.isHidden(v.context.$implicit);
    ck(v,3,0,currVal_2);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.isDisabled(v.context.$implicit);
    const currVal_1:any = co.isHidden(v.context.$implicit);
    ck(v,0,0,currVal_0,currVal_1);
  });
}
function View_NgbDatepickerMonthView_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[[
        'class',
        'ngb-dp-week d-flex'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbDatepickerMonthView_6)),
    import0.ɵdid(16384,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbDatepickerMonthView_7)),
    import0.ɵdid(802816,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.showWeekNumbers;
    ck(v,3,0,currVal_0);
    const currVal_1:any = (<any>v.parent).context.$implicit.days;
    ck(v,6,0,currVal_1);
  },(null as any));
}
function View_NgbDatepickerMonthView_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbDatepickerMonthView_5)),
    import0.ɵdid(16384,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:boolean = !co.isCollapsed(v.context.$implicit);
    ck(v,2,0,currVal_0);
  },(null as any));
}
export function View_NgbDatepickerMonthView_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbDatepickerMonthView_1)),
    import0.ɵdid(16384,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbDatepickerMonthView_4)),
    import0.ɵdid(802816,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n  ']))
  ]
  ,(ck,v) => {
    var co:import2.NgbDatepickerMonthView = v.component;
    const currVal_0:any = co.showWeekdays;
    ck(v,2,0,currVal_0);
    const currVal_1:any = co.month.weeks;
    ck(v,5,0,currVal_1);
  },(null as any));
}
function View_NgbDatepickerMonthView_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ngb-datepicker-month-view',[[
        'class',
        'd-block'
      ]
    ],(null as any),(null as any),(null as any),View_NgbDatepickerMonthView_0,RenderType_NgbDatepickerMonthView)),
    import0.ɵdid(49152,(null as any),0,import2.NgbDatepickerMonthView,[import3.NgbDatepickerI18n],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const NgbDatepickerMonthViewNgFactory:import0.ComponentFactory<import2.NgbDatepickerMonthView> = import0.ɵccf('ngb-datepicker-month-view',import2.NgbDatepickerMonthView,View_NgbDatepickerMonthView_Host_0,{
  dayTemplate: 'dayTemplate',
  disabled: 'disabled',
  month: 'month',
  outsideDays: 'outsideDays',
  selectedDate: 'selectedDate',
  showWeekdays: 'showWeekdays',
  showWeekNumbers: 'showWeekNumbers'
}
,{select: 'select'},([] as any[]));
