// @ng-bootstrap/ng-bootstrap/modal/modal-window.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@ng-bootstrap/ng-bootstrap/modal/modal-window';
const styles_NgbModalWindow:any[] = ([] as any[]);
export const RenderType_NgbModalWindow:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_NgbModalWindow,
  data: {}
}
);
export function View_NgbModalWindow_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),4,'div',[[
        'role',
        'document'
      ]
      ],[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'modal-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(null as any),(ck,v) => {
    var co:import1.NgbModalWindow = v.component;
    const currVal_0:any = ('modal-dialog' + (co.size? (' modal-' + co.size): ''));
    ck(v,1,0,currVal_0);
  });
}
function View_NgbModalWindow_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ngb-modal-window',[
      [
        'role',
        'dialog'
      ]
      ,
      [
        'style',
        'display: block;'
      ]
      ,
      [
        'tabindex',
        '-1'
      ]

    ]
      ,[[
        8,
        'className',
        0
      ]
    ],[
      [
        (null as any),
        'keyup.esc'
      ]
      ,
      [
        (null as any),
        'click'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('keyup.esc' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).escKey($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('click' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,1).backdropClick($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },View_NgbModalWindow_0,RenderType_NgbModalWindow)),
    import0.ɵdid(4440064,(null as any),0,import1.NgbModalWindow,[
      import0.ElementRef,
      import0.Renderer2
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = ('modal fade show' + (import0.ɵnov(v,1).windowClass? (' ' + import0.ɵnov(v,1).windowClass): ''));
    ck(v,0,0,currVal_0);
  });
}
export const NgbModalWindowNgFactory:import0.ComponentFactory<import1.NgbModalWindow> = import0.ɵccf('ngb-modal-window',import1.NgbModalWindow,View_NgbModalWindow_Host_0,{
  backdrop: 'backdrop',
  keyboard: 'keyboard',
  size: 'size',
  windowClass: 'windowClass'
}
,{dismissEvent: 'dismiss'},['*']);
