// @ng-bootstrap/ng-bootstrap/modal/modal-backdrop.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@ng-bootstrap/ng-bootstrap/modal/modal-backdrop';
const styles_NgbModalBackdrop:any[] = ([] as any[]);
export const RenderType_NgbModalBackdrop:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_NgbModalBackdrop,
  data: {}
}
);
export function View_NgbModalBackdrop_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_NgbModalBackdrop_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ngb-modal-backdrop',[[
        'class',
        'modal-backdrop fade show'
      ]
    ],(null as any),(null as any),(null as any),View_NgbModalBackdrop_0,RenderType_NgbModalBackdrop)),
    import0.ɵdid(49152,(null as any),0,import1.NgbModalBackdrop,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const NgbModalBackdropNgFactory:import0.ComponentFactory<import1.NgbModalBackdrop> = import0.ɵccf('ngb-modal-backdrop',import1.NgbModalBackdrop,View_NgbModalBackdrop_Host_0,{},{},([] as any[]));
