// @ng-bootstrap/ng-bootstrap/tooltip/tooltip.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@ng-bootstrap/ng-bootstrap/tooltip/tooltip';
const styles_NgbTooltipWindow:any[] = ([] as any[]);
export const RenderType_NgbTooltipWindow:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_NgbTooltipWindow,
  data: {}
}
);
export function View_NgbTooltipWindow_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'tooltip-inner'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(null as any),(null as any));
}
function View_NgbTooltipWindow_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ngb-tooltip-window',[[
        'role',
        'tooltip'
      ]
    ],[
      [
        8,
        'className',
        0
      ]
      ,
      [
        8,
        'id',
        0
      ]

    ]
    ,(null as any),(null as any),View_NgbTooltipWindow_0,RenderType_NgbTooltipWindow)),
    import0.ɵdid(49152,(null as any),0,import1.NgbTooltipWindow,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = ('tooltip show tooltip-' + import0.ɵnov(v,1).placement);
    const currVal_1:any = import0.ɵnov(v,1).id;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const NgbTooltipWindowNgFactory:import0.ComponentFactory<import1.NgbTooltipWindow> = import0.ɵccf('ngb-tooltip-window',import1.NgbTooltipWindow,View_NgbTooltipWindow_Host_0,{
  placement: 'placement',
  id: 'id'
}
,{},['*']);
