// ng-jhipster/src/component/jhi-item-count.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '../language/jhi-translate.directive.ngfactory';
import * as import2 from 'ng-jhipster/src/language/jhi-translate.directive';
import * as import3 from '@angular/common';
import * as import4 from 'ng-jhipster/src/component/jhi-item-count.component';
import * as import5 from 'ng-jhipster/src/config.service';
const styles_JhiItemCountComponent:any[] = ([] as any[]);
export const RenderType_JhiItemCountComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_JhiItemCountComponent,
  data: {}
}
);
function View_JhiItemCountComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'div',[
      [
        'class',
        'info jhi-item-count'
      ]
      ,
      [
        'jhiTranslate',
        'global.item-count'
      ]

    ]
      ,[[
        1,
        'translateValues',
        0
      ]
    ],(null as any),(null as any),import1.View_JhiTranslateComponent_0,import1.RenderType_JhiTranslateComponent)),
    import0.ɵdid(49152,(null as any),0,import2.JhiTranslateComponent,([] as any[]),{
      jhiTranslate: [
        0,
        'jhiTranslate'
      ]
      ,
      translateValues: [
        1,
        'translateValues'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['  /* [attr.translateValues] is used to get entire values in tests */\n        ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = 'global.item-count';
    const currVal_2:any = import0.ɵinlineInterpolate(1,'',co.i18nValues(),'');
    ck(v,1,0,currVal_1,currVal_2);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.i18nValues();
    ck(v,0,0,currVal_0);
  });
}
function View_JhiItemCountComponent_2(l:any):import0.ɵViewDefinition {
    return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),[
      '\n            Showing ',
      ' -\n            ',
      '\n            of ',
      ' items.\n        '
    ]
  ))],(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ((((co.page - 1) * co.itemsPerPage) == 0)? 1: (((co.page - 1) * co.itemsPerPage) + 1));
    const currVal_1:any = (((co.page * co.itemsPerPage) < co.total)? (co.page * co.itemsPerPage): co.total);
    const currVal_2:any = co.total;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export function View_JhiItemCountComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_JhiItemCountComponent_1)),
    import0.ɵdid(16384,(null as any),0,import3.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
    ,{
      ngIf: [
        0,
        'ngIf'
      ]
      ,
      ngIfElse: [
        1,
        'ngIfElse'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵand(0,[[
        'noI18n',
        2
      ]
    ],(null as any),0,(null as any),View_JhiItemCountComponent_2))
  ]
  ,(ck,v) => {
    var co:import4.JhiItemCountComponent = v.component;
    const currVal_0:any = co.i18nEnabled;
    const currVal_1:any = import0.ɵnov(v,4);
    ck(v,2,0,currVal_0,currVal_1);
  },(null as any));
}
function View_JhiItemCountComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'jhi-item-count',([] as any[]),(null as any),(null as any),(null as any),View_JhiItemCountComponent_0,RenderType_JhiItemCountComponent)),
    import0.ɵdid(49152,(null as any),0,import4.JhiItemCountComponent,[import5.JhiConfigService],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const JhiItemCountComponentNgFactory:import0.ComponentFactory<import4.JhiItemCountComponent> = import0.ɵccf('jhi-item-count',import4.JhiItemCountComponent,View_JhiItemCountComponent_Host_0,{
  page: 'page',
  total: 'total',
  itemsPerPage: 'itemsPerPage'
}
,{},([] as any[]));
