// @ng-bootstrap/ng-bootstrap/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@ng-bootstrap/ng-bootstrap/index';
import * as import2 from '@angular/common';
import * as import3 from '@ng-bootstrap/ng-bootstrap/alert/alert.module';
import * as import4 from '@ng-bootstrap/ng-bootstrap/buttons/radio.module';
import * as import5 from '@ng-bootstrap/ng-bootstrap/collapse/collapse.module';
import * as import6 from '@ng-bootstrap/ng-bootstrap/progressbar/progressbar.module';
import * as import7 from '@ng-bootstrap/ng-bootstrap/tooltip/tooltip.module';
import * as import8 from '@ng-bootstrap/ng-bootstrap/typeahead/typeahead.module';
import * as import9 from '@ng-bootstrap/ng-bootstrap/accordion/accordion.module';
import * as import10 from '@ng-bootstrap/ng-bootstrap/carousel/carousel.module';
import * as import11 from '@angular/forms';
import * as import12 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker.module';
import * as import13 from '@ng-bootstrap/ng-bootstrap/dropdown/dropdown.module';
import * as import14 from '@ng-bootstrap/ng-bootstrap/modal/modal.module';
import * as import15 from '@ng-bootstrap/ng-bootstrap/pagination/pagination.module';
import * as import16 from '@ng-bootstrap/ng-bootstrap/popover/popover.module';
import * as import17 from '@ng-bootstrap/ng-bootstrap/rating/rating.module';
import * as import18 from '@ng-bootstrap/ng-bootstrap/tabset/tabset.module';
import * as import19 from '@ng-bootstrap/ng-bootstrap/timepicker/timepicker.module';
import * as import20 from '@ng-bootstrap/ng-bootstrap/modal/modal-stack';
import * as import21 from '@ng-bootstrap/ng-bootstrap/modal/modal';
import * as import22 from '@ng-bootstrap/ng-bootstrap/alert/alert-config';
import * as import23 from '@ng-bootstrap/ng-bootstrap/progressbar/progressbar-config';
import * as import24 from '@ng-bootstrap/ng-bootstrap/tooltip/tooltip-config';
import * as import25 from '@ng-bootstrap/ng-bootstrap/typeahead/typeahead-config';
import * as import26 from '@ng-bootstrap/ng-bootstrap/accordion/accordion-config';
import * as import27 from '@ng-bootstrap/ng-bootstrap/carousel/carousel-config';
import * as import28 from '@ng-bootstrap/ng-bootstrap/datepicker/ngb-calendar';
import * as import29 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker-i18n';
import * as import30 from '@ng-bootstrap/ng-bootstrap/datepicker/ngb-date-parser-formatter';
import * as import31 from '@ng-bootstrap/ng-bootstrap/datepicker/datepicker-config';
import * as import32 from '@ng-bootstrap/ng-bootstrap/dropdown/dropdown-config';
import * as import33 from '@ng-bootstrap/ng-bootstrap/pagination/pagination-config';
import * as import34 from '@ng-bootstrap/ng-bootstrap/popover/popover-config';
import * as import35 from '@ng-bootstrap/ng-bootstrap/rating/rating-config';
import * as import36 from '@ng-bootstrap/ng-bootstrap/tabset/tabset-config';
import * as import37 from '@ng-bootstrap/ng-bootstrap/timepicker/timepicker-config';
import * as import38 from './alert/alert.ngfactory';
import * as import39 from './tooltip/tooltip.ngfactory';
import * as import40 from './typeahead/typeahead-window.ngfactory';
import * as import41 from './datepicker/datepicker.ngfactory';
import * as import42 from './modal/modal-backdrop.ngfactory';
import * as import43 from './modal/modal-window.ngfactory';
import * as import44 from './popover/popover.ngfactory';
class NgbRootModuleInjector extends import0.ɵNgModuleInjector<import1.NgbRootModule> {
  _CommonModule_0:import2.CommonModule;
  _NgbAlertModule_1:import3.NgbAlertModule;
  _NgbButtonsModule_2:import4.NgbButtonsModule;
  _NgbCollapseModule_3:import5.NgbCollapseModule;
  _NgbProgressbarModule_4:import6.NgbProgressbarModule;
  _NgbTooltipModule_5:import7.NgbTooltipModule;
  _NgbTypeaheadModule_6:import8.NgbTypeaheadModule;
  _NgbAccordionModule_7:import9.NgbAccordionModule;
  _NgbCarouselModule_8:import10.NgbCarouselModule;
  _ɵba_9:import11.ɵba;
  _FormsModule_10:import11.FormsModule;
  _NgbDatepickerModule_11:import12.NgbDatepickerModule;
  _NgbDropdownModule_12:import13.NgbDropdownModule;
  _NgbModalModule_13:import14.NgbModalModule;
  _NgbPaginationModule_14:import15.NgbPaginationModule;
  _NgbPopoverModule_15:import16.NgbPopoverModule;
  _NgbRatingModule_16:import17.NgbRatingModule;
  _NgbTabsetModule_17:import18.NgbTabsetModule;
  _NgbTimepickerModule_18:import19.NgbTimepickerModule;
  _NgbRootModule_19:import1.NgbRootModule;
  __NgLocalization_20:import2.NgLocaleLocalization;
  __ɵi_21:import11.ɵi;
  __NgbModalStack_22:import20.NgbModalStack;
  __NgbModal_23:import21.NgbModal;
  __NgbAlertConfig_24:import22.NgbAlertConfig;
  __NgbProgressbarConfig_25:import23.NgbProgressbarConfig;
  __NgbTooltipConfig_26:import24.NgbTooltipConfig;
  __NgbTypeaheadConfig_27:import25.NgbTypeaheadConfig;
  __NgbAccordionConfig_28:import26.NgbAccordionConfig;
  __NgbCarouselConfig_29:import27.NgbCarouselConfig;
  __NgbCalendar_30:import28.NgbCalendarGregorian;
  __NgbDatepickerI18n_31:import29.NgbDatepickerI18nDefault;
  __NgbDateParserFormatter_32:import30.NgbDateISOParserFormatter;
  __NgbDatepickerConfig_33:import31.NgbDatepickerConfig;
  __NgbDropdownConfig_34:import32.NgbDropdownConfig;
  __NgbPaginationConfig_35:import33.NgbPaginationConfig;
  __NgbPopoverConfig_36:import34.NgbPopoverConfig;
  __NgbRatingConfig_37:import35.NgbRatingConfig;
  __NgbTabsetConfig_38:import36.NgbTabsetConfig;
  __NgbTimepickerConfig_39:import37.NgbTimepickerConfig;
  constructor(parent:import0.Injector) {
    super(parent,[
      import38.NgbAlertNgFactory,
      import39.NgbTooltipWindowNgFactory,
      import40.NgbTypeaheadWindowNgFactory,
      import41.NgbDatepickerNgFactory,
      import42.NgbModalBackdropNgFactory,
      import43.NgbModalWindowNgFactory,
      import44.NgbPopoverWindowNgFactory
    ]
    ,([] as any[]));
  }
  get _NgLocalization_20():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_20 == null)) { (this.__NgLocalization_20 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_20;
  }
  get _ɵi_21():import11.ɵi {
    if ((this.__ɵi_21 == null)) { (this.__ɵi_21 = new import11.ɵi()); }
    return this.__ɵi_21;
  }
  get _NgbModalStack_22():import20.NgbModalStack {
    if ((this.__NgbModalStack_22 == null)) { (this.__NgbModalStack_22 = new import20.NgbModalStack(this.parent.get(import0.ApplicationRef),this,this.componentFactoryResolver)); }
    return this.__NgbModalStack_22;
  }
  get _NgbModal_23():import21.NgbModal {
    if ((this.__NgbModal_23 == null)) { (this.__NgbModal_23 = new import21.NgbModal(this.componentFactoryResolver,this,this._NgbModalStack_22)); }
    return this.__NgbModal_23;
  }
  get _NgbAlertConfig_24():import22.NgbAlertConfig {
    if ((this.__NgbAlertConfig_24 == null)) { (this.__NgbAlertConfig_24 = new import22.NgbAlertConfig()); }
    return this.__NgbAlertConfig_24;
  }
  get _NgbProgressbarConfig_25():import23.NgbProgressbarConfig {
    if ((this.__NgbProgressbarConfig_25 == null)) { (this.__NgbProgressbarConfig_25 = new import23.NgbProgressbarConfig()); }
    return this.__NgbProgressbarConfig_25;
  }
  get _NgbTooltipConfig_26():import24.NgbTooltipConfig {
    if ((this.__NgbTooltipConfig_26 == null)) { (this.__NgbTooltipConfig_26 = new import24.NgbTooltipConfig()); }
    return this.__NgbTooltipConfig_26;
  }
  get _NgbTypeaheadConfig_27():import25.NgbTypeaheadConfig {
    if ((this.__NgbTypeaheadConfig_27 == null)) { (this.__NgbTypeaheadConfig_27 = new import25.NgbTypeaheadConfig()); }
    return this.__NgbTypeaheadConfig_27;
  }
  get _NgbAccordionConfig_28():import26.NgbAccordionConfig {
    if ((this.__NgbAccordionConfig_28 == null)) { (this.__NgbAccordionConfig_28 = new import26.NgbAccordionConfig()); }
    return this.__NgbAccordionConfig_28;
  }
  get _NgbCarouselConfig_29():import27.NgbCarouselConfig {
    if ((this.__NgbCarouselConfig_29 == null)) { (this.__NgbCarouselConfig_29 = new import27.NgbCarouselConfig()); }
    return this.__NgbCarouselConfig_29;
  }
  get _NgbCalendar_30():import28.NgbCalendarGregorian {
    if ((this.__NgbCalendar_30 == null)) { (this.__NgbCalendar_30 = new import28.NgbCalendarGregorian()); }
    return this.__NgbCalendar_30;
  }
  get _NgbDatepickerI18n_31():import29.NgbDatepickerI18nDefault {
    if ((this.__NgbDatepickerI18n_31 == null)) { (this.__NgbDatepickerI18n_31 = new import29.NgbDatepickerI18nDefault()); }
    return this.__NgbDatepickerI18n_31;
  }
  get _NgbDateParserFormatter_32():import30.NgbDateISOParserFormatter {
    if ((this.__NgbDateParserFormatter_32 == null)) { (this.__NgbDateParserFormatter_32 = new import30.NgbDateISOParserFormatter()); }
    return this.__NgbDateParserFormatter_32;
  }
  get _NgbDatepickerConfig_33():import31.NgbDatepickerConfig {
    if ((this.__NgbDatepickerConfig_33 == null)) { (this.__NgbDatepickerConfig_33 = new import31.NgbDatepickerConfig()); }
    return this.__NgbDatepickerConfig_33;
  }
  get _NgbDropdownConfig_34():import32.NgbDropdownConfig {
    if ((this.__NgbDropdownConfig_34 == null)) { (this.__NgbDropdownConfig_34 = new import32.NgbDropdownConfig()); }
    return this.__NgbDropdownConfig_34;
  }
  get _NgbPaginationConfig_35():import33.NgbPaginationConfig {
    if ((this.__NgbPaginationConfig_35 == null)) { (this.__NgbPaginationConfig_35 = new import33.NgbPaginationConfig()); }
    return this.__NgbPaginationConfig_35;
  }
  get _NgbPopoverConfig_36():import34.NgbPopoverConfig {
    if ((this.__NgbPopoverConfig_36 == null)) { (this.__NgbPopoverConfig_36 = new import34.NgbPopoverConfig()); }
    return this.__NgbPopoverConfig_36;
  }
  get _NgbRatingConfig_37():import35.NgbRatingConfig {
    if ((this.__NgbRatingConfig_37 == null)) { (this.__NgbRatingConfig_37 = new import35.NgbRatingConfig()); }
    return this.__NgbRatingConfig_37;
  }
  get _NgbTabsetConfig_38():import36.NgbTabsetConfig {
    if ((this.__NgbTabsetConfig_38 == null)) { (this.__NgbTabsetConfig_38 = new import36.NgbTabsetConfig()); }
    return this.__NgbTabsetConfig_38;
  }
  get _NgbTimepickerConfig_39():import37.NgbTimepickerConfig {
    if ((this.__NgbTimepickerConfig_39 == null)) { (this.__NgbTimepickerConfig_39 = new import37.NgbTimepickerConfig()); }
    return this.__NgbTimepickerConfig_39;
  }
  createInternal():import1.NgbRootModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._NgbAlertModule_1 = new import3.NgbAlertModule();
    this._NgbButtonsModule_2 = new import4.NgbButtonsModule();
    this._NgbCollapseModule_3 = new import5.NgbCollapseModule();
    this._NgbProgressbarModule_4 = new import6.NgbProgressbarModule();
    this._NgbTooltipModule_5 = new import7.NgbTooltipModule();
    this._NgbTypeaheadModule_6 = new import8.NgbTypeaheadModule();
    this._NgbAccordionModule_7 = new import9.NgbAccordionModule();
    this._NgbCarouselModule_8 = new import10.NgbCarouselModule();
    this._ɵba_9 = new import11.ɵba();
    this._FormsModule_10 = new import11.FormsModule();
    this._NgbDatepickerModule_11 = new import12.NgbDatepickerModule();
    this._NgbDropdownModule_12 = new import13.NgbDropdownModule();
    this._NgbModalModule_13 = new import14.NgbModalModule();
    this._NgbPaginationModule_14 = new import15.NgbPaginationModule();
    this._NgbPopoverModule_15 = new import16.NgbPopoverModule();
    this._NgbRatingModule_16 = new import17.NgbRatingModule();
    this._NgbTabsetModule_17 = new import18.NgbTabsetModule();
    this._NgbTimepickerModule_18 = new import19.NgbTimepickerModule();
    this._NgbRootModule_19 = new import1.NgbRootModule();
    return this._NgbRootModule_19;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.NgbAlertModule)) { return this._NgbAlertModule_1; }
    if ((token === import4.NgbButtonsModule)) { return this._NgbButtonsModule_2; }
    if ((token === import5.NgbCollapseModule)) { return this._NgbCollapseModule_3; }
    if ((token === import6.NgbProgressbarModule)) { return this._NgbProgressbarModule_4; }
    if ((token === import7.NgbTooltipModule)) { return this._NgbTooltipModule_5; }
    if ((token === import8.NgbTypeaheadModule)) { return this._NgbTypeaheadModule_6; }
    if ((token === import9.NgbAccordionModule)) { return this._NgbAccordionModule_7; }
    if ((token === import10.NgbCarouselModule)) { return this._NgbCarouselModule_8; }
    if ((token === import11.ɵba)) { return this._ɵba_9; }
    if ((token === import11.FormsModule)) { return this._FormsModule_10; }
    if ((token === import12.NgbDatepickerModule)) { return this._NgbDatepickerModule_11; }
    if ((token === import13.NgbDropdownModule)) { return this._NgbDropdownModule_12; }
    if ((token === import14.NgbModalModule)) { return this._NgbModalModule_13; }
    if ((token === import15.NgbPaginationModule)) { return this._NgbPaginationModule_14; }
    if ((token === import16.NgbPopoverModule)) { return this._NgbPopoverModule_15; }
    if ((token === import17.NgbRatingModule)) { return this._NgbRatingModule_16; }
    if ((token === import18.NgbTabsetModule)) { return this._NgbTabsetModule_17; }
    if ((token === import19.NgbTimepickerModule)) { return this._NgbTimepickerModule_18; }
    if ((token === import1.NgbRootModule)) { return this._NgbRootModule_19; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_20; }
    if ((token === import11.ɵi)) { return this._ɵi_21; }
    if ((token === import20.NgbModalStack)) { return this._NgbModalStack_22; }
    if ((token === import21.NgbModal)) { return this._NgbModal_23; }
    if ((token === import22.NgbAlertConfig)) { return this._NgbAlertConfig_24; }
    if ((token === import23.NgbProgressbarConfig)) { return this._NgbProgressbarConfig_25; }
    if ((token === import24.NgbTooltipConfig)) { return this._NgbTooltipConfig_26; }
    if ((token === import25.NgbTypeaheadConfig)) { return this._NgbTypeaheadConfig_27; }
    if ((token === import26.NgbAccordionConfig)) { return this._NgbAccordionConfig_28; }
    if ((token === import27.NgbCarouselConfig)) { return this._NgbCarouselConfig_29; }
    if ((token === import28.NgbCalendar)) { return this._NgbCalendar_30; }
    if ((token === import29.NgbDatepickerI18n)) { return this._NgbDatepickerI18n_31; }
    if ((token === import30.NgbDateParserFormatter)) { return this._NgbDateParserFormatter_32; }
    if ((token === import31.NgbDatepickerConfig)) { return this._NgbDatepickerConfig_33; }
    if ((token === import32.NgbDropdownConfig)) { return this._NgbDropdownConfig_34; }
    if ((token === import33.NgbPaginationConfig)) { return this._NgbPaginationConfig_35; }
    if ((token === import34.NgbPopoverConfig)) { return this._NgbPopoverConfig_36; }
    if ((token === import35.NgbRatingConfig)) { return this._NgbRatingConfig_37; }
    if ((token === import36.NgbTabsetConfig)) { return this._NgbTabsetConfig_38; }
    if ((token === import37.NgbTimepickerConfig)) { return this._NgbTimepickerConfig_39; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const NgbRootModuleNgFactory:import0.NgModuleFactory<import1.NgbRootModule> = new import0.NgModuleFactory<any>(NgbRootModuleInjector,import1.NgbRootModule);
class NgbModuleInjector extends import0.ɵNgModuleInjector<import1.NgbModule> {
  _CommonModule_0:import2.CommonModule;
  _NgbAccordionModule_1:import9.NgbAccordionModule;
  _NgbAlertModule_2:import3.NgbAlertModule;
  _NgbButtonsModule_3:import4.NgbButtonsModule;
  _NgbCarouselModule_4:import10.NgbCarouselModule;
  _NgbCollapseModule_5:import5.NgbCollapseModule;
  _ɵba_6:import11.ɵba;
  _FormsModule_7:import11.FormsModule;
  _NgbDatepickerModule_8:import12.NgbDatepickerModule;
  _NgbDropdownModule_9:import13.NgbDropdownModule;
  _NgbModalModule_10:import14.NgbModalModule;
  _NgbPaginationModule_11:import15.NgbPaginationModule;
  _NgbPopoverModule_12:import16.NgbPopoverModule;
  _NgbProgressbarModule_13:import6.NgbProgressbarModule;
  _NgbRatingModule_14:import17.NgbRatingModule;
  _NgbTabsetModule_15:import18.NgbTabsetModule;
  _NgbTimepickerModule_16:import19.NgbTimepickerModule;
  _NgbTooltipModule_17:import7.NgbTooltipModule;
  _NgbTypeaheadModule_18:import8.NgbTypeaheadModule;
  _NgbModule_19:import1.NgbModule;
  __NgLocalization_20:import2.NgLocaleLocalization;
  __ɵi_21:import11.ɵi;
  __NgbModal_22:import21.NgbModal;
  constructor(parent:import0.Injector) {
    super(parent,[
      import38.NgbAlertNgFactory,
      import41.NgbDatepickerNgFactory,
      import42.NgbModalBackdropNgFactory,
      import43.NgbModalWindowNgFactory,
      import44.NgbPopoverWindowNgFactory,
      import39.NgbTooltipWindowNgFactory,
      import40.NgbTypeaheadWindowNgFactory
    ]
    ,([] as any[]));
  }
  get _NgLocalization_20():import2.NgLocaleLocalization {
    if ((this.__NgLocalization_20 == null)) { (this.__NgLocalization_20 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_20;
  }
  get _ɵi_21():import11.ɵi {
    if ((this.__ɵi_21 == null)) { (this.__ɵi_21 = new import11.ɵi()); }
    return this.__ɵi_21;
  }
  get _NgbModal_22():import21.NgbModal {
    if ((this.__NgbModal_22 == null)) { (this.__NgbModal_22 = new import21.NgbModal(this.componentFactoryResolver,this,this.parent.get(import20.NgbModalStack))); }
    return this.__NgbModal_22;
  }
  createInternal():import1.NgbModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._NgbAccordionModule_1 = new import9.NgbAccordionModule();
    this._NgbAlertModule_2 = new import3.NgbAlertModule();
    this._NgbButtonsModule_3 = new import4.NgbButtonsModule();
    this._NgbCarouselModule_4 = new import10.NgbCarouselModule();
    this._NgbCollapseModule_5 = new import5.NgbCollapseModule();
    this._ɵba_6 = new import11.ɵba();
    this._FormsModule_7 = new import11.FormsModule();
    this._NgbDatepickerModule_8 = new import12.NgbDatepickerModule();
    this._NgbDropdownModule_9 = new import13.NgbDropdownModule();
    this._NgbModalModule_10 = new import14.NgbModalModule();
    this._NgbPaginationModule_11 = new import15.NgbPaginationModule();
    this._NgbPopoverModule_12 = new import16.NgbPopoverModule();
    this._NgbProgressbarModule_13 = new import6.NgbProgressbarModule();
    this._NgbRatingModule_14 = new import17.NgbRatingModule();
    this._NgbTabsetModule_15 = new import18.NgbTabsetModule();
    this._NgbTimepickerModule_16 = new import19.NgbTimepickerModule();
    this._NgbTooltipModule_17 = new import7.NgbTooltipModule();
    this._NgbTypeaheadModule_18 = new import8.NgbTypeaheadModule();
    this._NgbModule_19 = new import1.NgbModule();
    return this._NgbModule_19;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import9.NgbAccordionModule)) { return this._NgbAccordionModule_1; }
    if ((token === import3.NgbAlertModule)) { return this._NgbAlertModule_2; }
    if ((token === import4.NgbButtonsModule)) { return this._NgbButtonsModule_3; }
    if ((token === import10.NgbCarouselModule)) { return this._NgbCarouselModule_4; }
    if ((token === import5.NgbCollapseModule)) { return this._NgbCollapseModule_5; }
    if ((token === import11.ɵba)) { return this._ɵba_6; }
    if ((token === import11.FormsModule)) { return this._FormsModule_7; }
    if ((token === import12.NgbDatepickerModule)) { return this._NgbDatepickerModule_8; }
    if ((token === import13.NgbDropdownModule)) { return this._NgbDropdownModule_9; }
    if ((token === import14.NgbModalModule)) { return this._NgbModalModule_10; }
    if ((token === import15.NgbPaginationModule)) { return this._NgbPaginationModule_11; }
    if ((token === import16.NgbPopoverModule)) { return this._NgbPopoverModule_12; }
    if ((token === import6.NgbProgressbarModule)) { return this._NgbProgressbarModule_13; }
    if ((token === import17.NgbRatingModule)) { return this._NgbRatingModule_14; }
    if ((token === import18.NgbTabsetModule)) { return this._NgbTabsetModule_15; }
    if ((token === import19.NgbTimepickerModule)) { return this._NgbTimepickerModule_16; }
    if ((token === import7.NgbTooltipModule)) { return this._NgbTooltipModule_17; }
    if ((token === import8.NgbTypeaheadModule)) { return this._NgbTypeaheadModule_18; }
    if ((token === import1.NgbModule)) { return this._NgbModule_19; }
    if ((token === import2.NgLocalization)) { return this._NgLocalization_20; }
    if ((token === import11.ɵi)) { return this._ɵi_21; }
    if ((token === import21.NgbModal)) { return this._NgbModal_22; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const NgbModuleNgFactory:import0.NgModuleFactory<import1.NgbModule> = new import0.NgModuleFactory<any>(NgbModuleInjector,import1.NgbModule);
