// @ng-bootstrap/ng-bootstrap/alert/alert.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from '@ng-bootstrap/ng-bootstrap/alert/alert';
import * as import3 from '@ng-bootstrap/ng-bootstrap/alert/alert-config';
const styles_NgbAlert:any[] = ([] as any[]);
export const RenderType_NgbAlert:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_NgbAlert,
  data: {}
}
);
function View_NgbAlert_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'button',[
      [
        'aria-label',
        'Close'
      ]
      ,
      [
        'class',
        'close'
      ]
      ,
      [
        'type',
        'button'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.closeHandler()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',[[
        'aria-hidden',
        'true'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['×'])),
    (l()(),import0.ɵted((null as any),['\n      ']))
  ]
  ,(null as any),(null as any));
}
export function View_NgbAlert_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',[[
        'role',
        'alert'
      ]
      ],[[
        8,
        'className',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n      '])),
    (l()(),import0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbAlert_1)),
    import0.ɵdid(16384,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n      '])),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n    '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import2.NgbAlert = v.component;
    const currVal_1:any = co.dismissible;
    ck(v,4,0,currVal_1);
  },(ck,v) => {
    var co:import2.NgbAlert = v.component;
    const currVal_0:any = (('alert alert-' + co.type) + (co.dismissible? ' alert-dismissible': ''));
    ck(v,1,0,currVal_0);
  });
}
function View_NgbAlert_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ngb-alert',([] as any[]),(null as any),(null as any),(null as any),View_NgbAlert_0,RenderType_NgbAlert)),
    import0.ɵdid(49152,(null as any),0,import2.NgbAlert,[import3.NgbAlertConfig],(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const NgbAlertNgFactory:import0.ComponentFactory<import2.NgbAlert> = import0.ɵccf('ngb-alert',import2.NgbAlert,View_NgbAlert_Host_0,{
  dismissible: 'dismissible',
  type: 'type'
}
,{close: 'close'},['*']);
