// ng-jhipster/src/language/jhi-translate.directive.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@ngx-translate/core/src/translate.pipe';
import * as import2 from '@ngx-translate/core/src/translate.service';
import * as import3 from 'ng-jhipster/src/language/jhi-translate.directive';
const styles_JhiTranslateComponent:any[] = ([] as any[]);
export const RenderType_JhiTranslateComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_JhiTranslateComponent,
  data: {}
}
);
export function View_JhiTranslateComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),[[
        8,
        'innerHTML',
        1
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import0.ɵpid(131072,import1.TranslatePipe,[
      import2.TranslateService,
      import0.ChangeDetectorRef
    ]
    )
  ]
  ,(null as any),(ck,v) => {
    var co:import3.JhiTranslateComponent = v.component;
    const currVal_0:any = import0.ɵunv(v,0,0,import0.ɵnov(v,1).transform(co.jhiTranslate,co.translateValues));
    ck(v,0,0,currVal_0);
  });
}
function View_JhiTranslateComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'jhiTranslate',
        ''
      ]
    ],(null as any),(null as any),(null as any),View_JhiTranslateComponent_0,RenderType_JhiTranslateComponent)),
      import0.ɵdid(49152,(null as any),0,import3.JhiTranslateComponent,([] as any[]),{jhiTranslate: [
        0,
        'jhiTranslate'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    const currVal_0:any = '';
    ck(v,1,0,currVal_0);
  },(null as any));
}
export const JhiTranslateComponentNgFactory:import0.ComponentFactory<import3.JhiTranslateComponent> = import0.ɵccf('[jhiTranslate]',import3.JhiTranslateComponent,View_JhiTranslateComponent_Host_0,{
  jhiTranslate: 'jhiTranslate',
  translateValues: 'translateValues'
}
,{},([] as any[]));
