// @ng-bootstrap/ng-bootstrap/popover/popover.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@ng-bootstrap/ng-bootstrap/popover/popover';
const styles_NgbPopoverWindow:any[] = ([] as any[]);
export const RenderType_NgbPopoverWindow:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_NgbPopoverWindow,
  data: {}
}
);
export function View_NgbPopoverWindow_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵted((null as any),['\n    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'h3',[[
        'class',
        'popover-title'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'popover-content'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(null as any),(ck,v) => {
    var co:import1.NgbPopoverWindow = v.component;
    const currVal_0:any = co.title;
    ck(v,2,0,currVal_0);
  });
}
function View_NgbPopoverWindow_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'ngb-popover-window',[[
        'role',
        'tooltip'
      ]
    ],[
      [
        8,
        'className',
        0
      ]
      ,
      [
        8,
        'id',
        0
      ]

    ]
    ,(null as any),(null as any),View_NgbPopoverWindow_0,RenderType_NgbPopoverWindow)),
    import0.ɵdid(49152,(null as any),0,import1.NgbPopoverWindow,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = ('popover show popover-' + import0.ɵnov(v,1).placement);
    const currVal_1:any = import0.ɵnov(v,1).id;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const NgbPopoverWindowNgFactory:import0.ComponentFactory<import1.NgbPopoverWindow> = import0.ɵccf('ngb-popover-window',import1.NgbPopoverWindow,View_NgbPopoverWindow_Host_0,{
  placement: 'placement',
  title: 'title',
  id: 'id'
}
,{},['*']);
